### PR TITLE
fix: replace `request` with `got` for HTTP calls

### PR DESCRIPTION
**Pull request by cristiand391**
### What does this PR do?

replace `request` dependency with `got`.
request is deprecated: https://www.npmjs.com/package/request

### What issues does this PR fix or reference?
@W-0@

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -2,6 +2,7 @@
   "extends": "@salesforce/dev-config/tsconfig",
   "compilerOptions": {
     "outDir": "lib",
+    "allowSyntheticDefaultImports": true,
```
Comment (cristiand391) [inline tsconfig.json:5]: wasn't able to build the plugin without this 🤷🏼 
https://github.com/TooTallNate/node-agent-base/issues/56

File: test/shared/installationVerification.test.ts [hunk 2]
```diff
@@ -475,29 +526,15 @@ describe('InstallationVerification Tests', () => {
...
-        cb(null, { statusCode: 200 } as request.Response, CERTIFICATE);
-      } else if (url.endsWith(plugin.name)) {
-        const err = new Error();
-        err.name = 'NPMMetaError';
-        cb(err, { statusCode: 500 } as request.Response, {});
-      } else {
-        throw new Error(`Unexpected test url - ${url}`);
-      }
-    };
-
```
Comment (cristiand391) [inline test/shared/installationVerification.test.ts:494]: I think this UT was from when plugin-trust was doing raw HTTP calls to npm, we switched to npm here:
https://github.com/salesforcecli/plugin-trust/pull/154

so no need to stub `got`, the error is thrown from here:
https://github.com/salesforcecli/plugin-trust/blob/1affd57bd25563be4278c16b9249c8d765475ca6/src/shared/npmCommand.ts#L189

File: test/shared/installationVerification.test.ts [hunk 3]
```diff
@@ -509,109 +546,6 @@ describe('InstallationVerification Tests', () => {
       });
   });
 
-  it.skip('server error', async () => {
```
Comment (cristiand391) [inline test/shared/installationVerification.test.ts:512]: ditto, this UT was testing the HTTP calls to npm, we run `npm show` now (see prev comment)

**Review by mshanemc — APPROVED** [review]

**Comment by mshanemc** [comment]
QA notes:

linked plugin-trust
✅ install a signed plugin (rel-mgmt), no warning
✅ install a signed plugin (org), no warning (did this because rel-mgmt installs its own plugin-trust, undoing my link, so I re-linked plugin-trust first)
✅ install an unsigned plugin (mine), got warning

not tested: proxies
not tested: a signed plugin with an invalid cert

**Pull request merged**